enhancements/ingress/aws-lb-eip-selection.md

```diff
@@ -0,0 +1,348 @@
+      - eipalloc-1111
+      - eipalloc-2222
+      - eipalloc-3333
+      - eipalloc-4444
+      - eipalloc-5555 
+...
+```  
+
+2. Installer creates a `default Ingress Controller CR` with the eipAllocations specified. 
+3. Ingress Operator will create the `default` `IngressController` with the service type load balancer having the service annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations` with values from the field `eipAllocations`
+   from the cluster ingress config object.
+
+#### Updating the default IngressController's EIPs after installation:
+
+The cluster-admin will need to update the eipAllocations in `IngressConfig` and then delete the `default IngressController`. 
+The cluster-ingress-operator will create a new `default IngressController` with the newly updated eipAllocations.
+
+### API Extensions
+
+This proposal doesn't add any API extensions other than the new field proposed in
+[IngressConfig API Updates](#ingressconfig-api-updates).
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+Currently, there is no unique consideration for making this change work with Hypershift.
+
+#### Standalone Clusters
+This proposal does not have any special implications for standalone clusters.
+
+#### Single-node Deployments or MicroShift
+This proposal does not have any special implications for single-node
+deployments or MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+#### Set EIP through installer for the default IngressController:
+    1. The admin sets the `eipAllocations` in the installer using the install-config.yaml.
+    2. The installer then creates a default Ingress Controller resource with the `eipAllocations`.
+    3. The cluster ingress operator watches the resource and then creates a `default` `IngressController` by:
+        1. creating an `default` `IngressController` from the `default` `Ingress Controller CR` created by the installer.
+        2. then scaling a deployment for the `IngressController` CR which has the value of the `eipAllocations`.
+        3. then creating a service of service type load balancer with the annotation `service.beta.kubernetes.io/aws-load-balancer-eip-allocations`,
+           which uses the value from the field `eipAllocations` of `IngressController` CR.
+
+#### Field Validation
+
+##### Validation on installer when installing in managed VPC (full-automated) based in the discovered zones used to create the cluster.
+   Add a function to the installer code to  compare the number of Availability Zones in the region to the number of eipAllocations passed in the `install-config.yaml`.   This is required because the cluster will select one subnet per AZ, so the number of EIPs must be equal to the number of subnets.
+
+##### The number of public subnets must match the number of `eipAllocations` in an unmanaged VPC (BYO VPC)
+     We will be comparing if the number of public subnets added to the install-config matches with len(eipAllocations).
```

Comment:
See https://github.com/openshift/installer/pull/8204#discussion_r1876602038, it's not quite this simple of a comparison.